src/main/scala/com/wavesplatform/state2/diffs/ExecuteContractFunctionTransactionDiff.scala

```diff
@@ -0,0 +1,31 @@
+package com.wavesplatform.state2.diffs
+
+//import cats.implicits._
+import com.wavesplatform.state2.{Diff, LeaseInfo, Portfolio}
+import com.wavesplatform.state2.reader.StateReader
+import scorex.transaction.ValidationError
+import scorex.transaction.ValidationError.GenericError
+import vsys.contract.ExecutionContext
+import vsys.state.opcdiffs.OpcFuncDiffer
+import vsys.transaction.contract.ExecuteContractFunctionTransaction
+import vsys.transaction.proof.{EllipticCurve25519Proof, Proofs}
+
+
+object ExecuteContractFunctionTransactionDiff {
+  def apply(s: StateReader, height: Int)(tx: ExecuteContractFunctionTransaction): Either[ValidationError, Diff] = {
+    if (tx.proofs.proofs.length > Proofs.MaxProofs) {
+      Left(GenericError(s"Too many proofs, max ${Proofs.MaxProofs} proofs"))
+    } else {
+      val sender = EllipticCurve25519Proof.fromBytes(tx.proofs.proofs.head.bytes.arr).toOption.get.publicKey
+      for {
+        exContext <- ExecutionContext.fromExeConTx(s, height, tx)
+        opcDiff <- OpcFuncDiffer(exContext)(tx.data)
+        diff = opcDiff.asTransactionDiff(height, tx)
+      } yield Diff(height = height, tx = tx,
+        portfolios = Map(sender.toAddress -> Portfolio(-tx.fee, LeaseInfo.empty, Map.empty)),
+        chargedFee = tx.fee
+      )
```

Comment:
should combine with diff here

Reply:
Will update this after the contractGen is finished. Right now only accept Diff without opcdiff